examples/AcousticWave2D_codegen.py

```diff
@@ -95,21 +95,43 @@ def _init_taylor(self, nt):
         # domain to the border
 
         # Forward wave equation
+
+        def damp_boundary(damp):
+            h = self.h
+            dampcoeff = 1.5 * np.log(1.0 / 0.001) / (40 * h)
+            nx, ny = self.model.get_shape()
+            nbpml = self.nbpml
+            for i in range(nbpml):
+                pos = np.abs((nbpml-i)/nbpml)
+                val = dampcoeff * (pos - np.sin(2*np.pi*pos)/(2*np.pi))
+                damp[i, :] += val
+                damp[-i, :] += val
+                damp[:, i] += val
+                damp[:, -i] += val
+
         wave_equation = m * dtt - (dxx + dyy) + e * dt
         stencil = solve(wave_equation, p(x, y, t+s))[0]
         self.nt = nt
-
+        m_sub = DenseData("m", self.model.vp.shape, self.dtype)
+        m_sub.initializer = lambda ref: np.copyto(ref, self.model.vp**(-2))
```

Comment:
Now that you ask me, I am not sure why I thought deferring was a good idea. I will change that. 

However, I don't think it would still become as simple as ``m_sub.data = self.model.vp**-2``. 

If I am not mistaken, a simple assignment like that is a pointer assignment. ``m_sub.data`` would just end up pointing to the memory allocated by ``self.model.vp``. Ideally we want to do something like ``initialize_model_vp(m_sub.data)``. Correct?